test/e2e/node_ip_migration.go

```diff
@@ -0,0 +1,584 @@
+		if utilnet.IsIPv6String(workerNodeIP) {
+			externalContainerIp = externalContainerIpv6
+		} else {
+			externalContainerIp = externalContainerIpv4
+		}
+		Expect(externalContainerIp).NotTo(BeEmpty())
+		framework.Logf("ExternalContainerIp is %s", externalContainerIp)
+
+		By("Creating a test pod on both selected worker nodes")
+		podWorkerNode := newAgnhostPodOnNode(
+			podWorkerNodeName,
+			workerNode,
+			podLabels,
+			podCommand...)
+		podSecondaryWorkerNode := newAgnhostPodOnNode(
+			podSecondaryWorkerNodeName,
+			secondaryWorkerNode,
+			podLabels,
+			podCommand...)
+		_ = fr.PodClient().CreateSync(podWorkerNode)
+		_ = fr.PodClient().CreateSync(podSecondaryWorkerNode)
+
+		By("Adding the \"k8s.ovn.org/egress-assignable\" label to the first node")
+		framework.AddOrUpdateLabelOnNode(fr.ClientSet, workerNode, "k8s.ovn.org/egress-assignable", "dummy")
+		podNamespace := fr.Namespace
+		podNamespace.Labels = map[string]string{
+			"name": fr.Namespace.Name,
+		}
+		updateNamespace(fr, podNamespace)
+
+		By("Creating an EgressIP object with one egress IP defined")
+		// Assign the egress IP without conflicting with any node IP,
+		// the kind subnet is /16 or /64 so the following should be fine.
+		// pick something at the end of the range to avoid conflicts with the kind / docker network setup.
+		// Exclude current ndoe IPs just for safety.
+		egressIp, err = findLastFreeSubnetIp(externalContainerName, externalContainerIp, []string{workerNodeIP, secondaryWorkerNodeIP})
+		framework.Logf("egressIP will be %s", egressIp)
+		Expect(err).NotTo(HaveOccurred())
+
+		var egressIPConfig = fmt.Sprintf(egressIPYamlTemplate, podLabels["app"], egressIp, "app", podLabels["app"], fr.Namespace.Name)
+		if err := ioutil.WriteFile(tmpDirIpMigration+"/"+egressIPYaml, []byte(egressIPConfig), 0644); err != nil {
+			framework.Failf("Unable to write CRD config to disk: %v", err)
+		}
+		_, err = framework.RunKubectl("default", "apply", "-f", tmpDirIpMigration+"/"+egressIPYaml)
+		Expect(err).NotTo(HaveOccurred())
+
+		By(fmt.Sprintf("Finding worker node %s's migration IP address", workerNode))
+		// pick something at the end of the range to avoid conflicts with the kind / docker network setup.
+		// the egress IP reachability check will not work here, so exclude it from the list; also exclude current node IPs just for safety.
+		nextWorkerNodeIP, err := findLastFreeSubnetIp(externalContainerName, externalContainerIp, []string{egressIp, workerNodeIP, secondaryWorkerNodeIP})
+		framework.Logf("New worker node IP will be %s", nextWorkerNodeIP)
+		Expect(err).NotTo(HaveOccurred())
+
+		By(fmt.Sprintf("Migrating worker node %s to IP address %s", workerNode, nextWorkerNodeIP))
+		err = migrateWorkerNodeIP(workerNode, workerNodeIP, nextWorkerNodeIP)
+		Expect(err).NotTo(HaveOccurred())
+
+		// It takes OVN up to 3 to 4 minutes to update the encap IP
```

Comment:
ovn-controller detecting the change in OVS? @dceara any ideas?

Reply:
Sounds like a bug if this is the case.  Can you please open an ovn-org/ovn issue with more information on how to replicate this?